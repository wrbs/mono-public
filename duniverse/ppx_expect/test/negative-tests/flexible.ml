open! Core

(*
   In old versions of [ppx_expect], the below tests would respect the formatting of the
   incorrect output present when possible.

   Currently, [ppx_expect] instead enforces standardized formatting in [[%expect]] nodes,
   so we instead use this test to demonstrate that all of the below expectations are
   reformatted to the same thing.
*)

(* Single line actual... *)

let%expect_test _ =
  print_string "hello";
  [%expect {||}]
;;

let%expect_test _ =
  print_string "hello";
  [%expect {|
  |}]
;;

let%expect_test _ =
  print_string "hello";
  [%expect {|
           |}]
;;

let%expect_test _ =
  print_string "hello";
  [%expect {|  WRONG
           |}]
;;

let%expect_test _ =
  print_string "hello";
  [%expect {|  WRONG
           |}]
;;

let%expect_test _ =
  print_string "hello";
  [%expect {|
  WRONG |}]
;;

let%expect_test _ =
  print_string "hello";
  [%expect {|
       WRONG
  |}]
;;

(* Multi line actual... *)

let%expect_test _ =
  print_string "one1\ntwo";
  [%expect {||}]
;;

let%expect_test _ =
  print_string "one2\ntwo";
  [%expect {|
  |}]
;;

let%expect_test _ =
  print_string "one3\ntwo";
  [%expect {|
           |}]
;;

let%expect_test _ =
  print_string "one4\ntwo";
  [%expect {|  WRONG
           |}]
;;

let%expect_test _ =
  print_string "one5\ntwo";
  [%expect {|
  WRONG |}]
;;

let%expect_test _ =
  print_string "one6\ntwo";
  [%expect {|
       WRONG
  |}]
;;

let%expect_test _ =
  print_string "one8\ntwo";
  [%expect {|
  WRONG
  THING |}]
;;

let%expect_test _ =
  print_string "one9\ntwo";
  [%expect {|
       WRONG
       THING
  |}]
;;

let%expect_test _ =
  print_string "one10\ntwo";
  [%expect {|
       WRONG
          THING
  |}]
;;

let%expect_test _ =
  print_string "one11\ntwo";
  [%expect {|
          WRONG
       THING
  |}]
;;
