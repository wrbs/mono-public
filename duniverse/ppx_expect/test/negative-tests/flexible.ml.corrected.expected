open! Core

(*
   In old versions of [ppx_expect], the below tests would respect the formatting of the
   incorrect output present when possible.

   Currently, [ppx_expect] instead enforces standardized formatting in [[%expect]] nodes,
   so we instead use this test to demonstrate that all of the below expectations are
   reformatted to the same thing.
*)

(* Single line actual... *)

let%expect_test _ =
  print_string "hello";
  [%expect {| hello |}]
;;

let%expect_test _ =
  print_string "hello";
  [%expect {| hello |}]
;;

let%expect_test _ =
  print_string "hello";
  [%expect {| hello |}]
;;

let%expect_test _ =
  print_string "hello";
  [%expect {| hello |}]
;;

let%expect_test _ =
  print_string "hello";
  [%expect {| hello |}]
;;

let%expect_test _ =
  print_string "hello";
  [%expect {| hello |}]
;;

let%expect_test _ =
  print_string "hello";
  [%expect {| hello |}]
;;

(* Multi line actual... *)

let%expect_test _ =
  print_string "one1\ntwo";
  [%expect
    {|
    one1
    two
    |}]
;;

let%expect_test _ =
  print_string "one2\ntwo";
  [%expect
    {|
    one2
    two
    |}]
;;

let%expect_test _ =
  print_string "one3\ntwo";
  [%expect
    {|
    one3
    two
    |}]
;;

let%expect_test _ =
  print_string "one4\ntwo";
  [%expect
    {|
    one4
    two
    |}]
;;

let%expect_test _ =
  print_string "one5\ntwo";
  [%expect
    {|
    one5
    two
    |}]
;;

let%expect_test _ =
  print_string "one6\ntwo";
  [%expect
    {|
    one6
    two
    |}]
;;

let%expect_test _ =
  print_string "one8\ntwo";
  [%expect
    {|
    one8
    two
    |}]
;;

let%expect_test _ =
  print_string "one9\ntwo";
  [%expect
    {|
    one9
    two
    |}]
;;

let%expect_test _ =
  print_string "one10\ntwo";
  [%expect
    {|
    one10
    two
    |}]
;;

let%expect_test _ =
  print_string "one11\ntwo";
  [%expect
    {|
    one11
    two
    |}]
;;
